### PR TITLE
feat: Model & schema validations for donations

### LINKED ISSUE #6061: feat: Max/Min price fields for donation tickets
**Describe the feature**
There is a requirement of min. price and max.price for donation tickets. Schemas need to be added as well.
Frontend PR: https://github.com/fossasia/open-event-frontend/pull/3069

**Additional info**
Working on it

### PR DESCRIPTION
**Pull request by mrsaicharan1**
Fixes #6061

#### Checklist

- [ ] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [ ] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:

- Adds schemas & fields for donations
#### Changes proposed in this pull request:

- Migration file
- Schema & model additions
- Added validation checks

### CODE REVIEW

**Comment by mrsaicharan1** [comment]
@uds5501 @shreyanshdwivedi Please review.

File: migrations/versions/43e8c59337ae_.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+# revision identifiers, used by Alembic.
+revision = '43e8c59337ae'
+down_revision = '14e31b3a860b'
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.add_column('tickets', sa.Column('max_price', sa.Float(), nullable=True))
+    op.add_column('tickets', sa.Column('min_price', sa.Float(), nullable=True))
```
Comment (uds5501) [inline migrations/versions/43e8c59337ae_.py]: @mrsaicharan1 Minimum price for a donation ticket should not be `null`, right?

**Comment by mrsaicharan1** [comment]
> @mrsaicharan1 I think there should be a validation check so that `min_price <= max_price`.

Yeah. Made it.

**Comment by mrsaicharan1** [comment]
@uds5501 @shreyanshdwivedi @iamareebjamal Please review.

File: app/api/schema/tickets.py [hunk 2]
```diff
@@ -57,6 +57,11 @@ def validate_quantity(self, data):
                 raise UnprocessableEntity({'pointer': '/data/attributes/quantity'},
                                           "quantity should be greater than or equal to min-order")
 
+        if 'min_price' in data and 'max_price' in data:
```
Comment (uds5501) [inline app/api/schema/tickets.py]: A corresponding validation test can be added regarding this.

File: app/api/schema/tickets.py [hunk 3]
```diff
@@ -57,6 +57,11 @@ def validate_quantity(self, data):
...
+                if data['min_price'] > data['max_price']:
+                    raise UnprocessableEntity({'pointer': '/data/attributes/quantity'},
```
Comment (shreyanshdwivedi) [inline app/api/schema/tickets.py]: @mrsaicharan1 I think you forgot to update pointer here

Comment (CosmicCoder96) [inline app/api/schema/tickets.py]: This exception logic should only apply for the donation tickets, type check for them too.

File: migrations/versions/43e8c59337ae_.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+
+# revision identifiers, used by Alembic.
+revision = '43e8c59337ae'
+down_revision = '14e31b3a860b'
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.add_column('tickets', sa.Column('max_price', sa.Float(), nullable=True))
```
Comment (CosmicCoder96) [inline migrations/versions/43e8c59337ae_.py]: here it's nullable false, I don't see any specification in the model file.

File: migrations/versions/43e8c59337ae_.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    op.add_column('tickets', sa.Column('min_price', sa.Float(), nullable=False))
```
Comment (CosmicCoder96) [inline migrations/versions/43e8c59337ae_.py]: here it's nullable false, I don't see any specification in the model file.

File: app/api/schema/tickets.py [hunk 6]
```diff
@@ -78,6 +83,8 @@ def validate_discount_code(self, data, original_data):
     description = fields.Str(allow_none=True)
     type = fields.Str(required=True)
     price = fields.Float(validate=lambda n: n >= 0, allow_none=True)
+    min_price = fields.Float(validate=lambda n: n >= 0, allow_none=True)
```
Comment (CosmicCoder96) [inline app/api/schema/tickets.py]: if nullable is false, why would you allow none here?

Comment (mrsaicharan1) [reply app/api/schema/tickets.py]: Yeah, noticed that. I'm updating it right now.

**Comment by mrsaicharan1** [comment]
@CosmicCoder96 I've made the requested changes. Please review.

File: app/models/ticket.py [hunk 7]
```diff
@@ -32,6 +32,8 @@ class Ticket(SoftDeletionModel):
     quantity = db.Column(db.Integer)
     position = db.Column(db.Integer)
     price = db.Column(db.Float)
+    min_price = db.Column(db.Float, nullable=False)
```
Comment (iamareebjamal) [inline app/models/ticket.py]: Provide a default

File: migrations/versions/43e8c59337ae_.py [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+
+# revision identifiers, used by Alembic.
+revision = '43e8c59337ae'
+down_revision = '232b13d4e86a'
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.add_column('tickets', sa.Column('max_price', sa.Float(), nullable=True))
+    op.add_column('tickets', sa.Column('min_price', sa.Float(), nullable=False))
```
Comment (iamareebjamal) [inline migrations/versions/43e8c59337ae_.py]: Provide a default

Comment (mrsaicharan1) [reply migrations/versions/43e8c59337ae_.py]: @iamareebjamal Oh okay! I thought only boolean fields required a `server_default`. Understood now.

**Comment by mrsaicharan1** [comment]
@iamareebjamal Oh okay! I thought only boolean fields required a server_default. Understood now. Updated it.

File: migrations/versions/43e8c59337ae_.py [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+
+
+# revision identifiers, used by Alembic.
+revision = '43e8c59337ae'
+down_revision = '232b13d4e86a'
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.add_column('tickets', sa.Column('max_price', sa.Float(), server_default=0, nullable=True))
```
Comment (iamareebjamal) [inline migrations/versions/43e8c59337ae_.py]: This is nullable, so no default is recommended

Comment (mrsaicharan1) [reply migrations/versions/43e8c59337ae_.py]: @iamareebjamal Done.

File: app/models/ticket.py (hunk 7 above)
Comment (iamareebjamal) [reply app/models/ticket.py]: Write a default here as well

**Pull request merged**